### PR TITLE
use mage in windows test runner

### LINKED ISSUE #1248: Fix Windows tests
Windows system tests are not running, report is still success. 

eg: https://apm-ci.elastic.co/job/elastic+apm-server+master+multijob-windows/beat=apm-server,label=windows/225/console
```
09:08:51 nosetests : The term 'nosetests' is not recognized as the name of a cmdlet, function, script file, or operable 
09:08:51 program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
09:08:51 At C:\Users\jenkins\workspace\elastic+apm-server+master+multijob-windows\beat\apm-server\label\windows\src\github.com\e
09:08:51 lastic\apm-server\script\jenkins\ci.ps1:48 char:8
```

We should also reuse as much as possible Beats scripting code, to get all changes on building fields, etc.

That is done in https://github.com/elastic/apm-server/pull/1247 for the other tests, but for windows is  more complicated.

### PR DESCRIPTION
**Pull request by graphaelli**
uses new capabilities introduced in elastic/beats#7766

closes #1248

### CODE REVIEW

**Comment by jalvz** [comment]
Thanks @graphaelli for attempting this!

Issue seems to persist: https://apm-ci.elastic.co/job/elastic+apm-server+pull-request+multijob-windows/beat=apm-server,label=windows/1197/console

```
08:09:05 nosetests : The term 'nosetests' is not recognized as the name of a cmdlet, function, script file, or operable 
08:09:05 program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
08:09:05 At C:\Users\jenkins\workspace\elastic+apm-server+pull-request+multijob-windows\beat\apm-server\label\windows\src\github
08:09:05 .com\elastic\apm-server\script\jenkins\ci.ps1:48 char:8
08:09:05 + exec { nosetests --with-timer --with-xunit --xunit-file=../../build/T ...
08:09:05 +        ~~~~~~~~~
08:09:05     + CategoryInfo          : ObjectNotFound: (nosetests:String) [], CommandNotFoundException
08:09:05     + FullyQualifiedErrorId : CommandNotFoundException
```

**Comment by graphaelli** [comment]
@jalvz Sorry, I should have been more clear - this requires some changes to our jenkins infrastructure as well.  I have not been able to figure out how to make these fail when `ci.ps1` exits with failure.

File: script/jenkins/ci.ps1 [hunk 1]
```diff
@@ -1,48 +1,61 @@
...
-exec { Get-Content build/TEST-go-unit.out | go-junit-report.exe -set-exit-code | Out-File -encoding UTF8 build/TEST-go-unit.xml } "Unit test FAILURE"
+exec { mage goTestUnit }
 
 echo "System testing $env:beat"
-exec { go test -race -c -cover -covermode=atomic -coverpkg ./... }
-exec { cd tests/system }
+# Get a CSV list of package names.
+$packages = $(go list ./... | select-string -Pattern "/vendor/" -NotMatch | select-string -Pattern "/scripts/cmd/" -NotMatch)
+$packages = ($packages|group|Select -ExpandProperty Name) -join ","
+exec { go test -race -c -cover -covermode=atomic -coverpkg $packages } "go test FAILURE"
```
Comment (graphaelli) [inline script/jenkins/ci.ps1]: is this duplicating `mage goTestUnit` ?

Comment (graphaelli) [reply script/jenkins/ci.ps1]: it's similar but not identical `mage -v goTestUnit` runs `go test -v ./...`

**Comment by graphaelli** [comment]
jenkins, retest this please.  The jenkins infrastructure updates have been deployed.  Both issues should be resolved now.

**Comment by simitt** [comment]
jenkins, retest this please.

**Comment by jalvz** [comment]
the windows test is actually broken. if the goal of this pr is just to return the correct code, we can merge this and create a new issue for the failing test it if you prefer.

**Pull request merged**